### ISSUE TITLE
Add `core release list` and `core release get` commands

### ISSUE THREAD

**Issue by schlessera** [wp-cli/core-command#108]
It would be useful to be able to easily query the available releases of WordPress core through the .org API and also get the download URLs for a given update.

A use case I'm currently working on that this would benefit is https://github.com/wp-cli/wp-cli-tests/issues/51

**Comment by schlessera** [comment]
It doesn't seem to be possible to query for arbitrary patch version through the WordPress.org API. Am I missing something, like an extra query argument or a different endpoint?

**Comment by schlessera** [comment]
Also, the `locale` query for the API argument behave in an unexpected way as well. It does only produce a different result for the first (i.e. the "latest") release, all others seem to still point to `en_US` downloads...

**Comment by swissspidy** [comment]
> Also, the locale query for the API argument behave in an unexpected way as well. It does only produce a different result for the first (i.e. the "latest") release, all others seem to still point to en_US downloads...

Maybe @ocean90 knows more here?

**Comment by schlessera** [comment]
Doesn't he always do?

**Comment by ocean90** [comment]
Can you tell me what endoints you are trying? What's the request looking like? To retrieve all available versions you could use https://api.wordpress.org/core/stable-check/1.0/.

**Comment by swissspidy** [comment]
The current code uses `https://api.wordpress.org/core/version-check/1.7/` it seems.

**Comment by schlessera** [comment]
Yes, indeed, I'm using the `version-check` endpoint right now. If I have a specific version from `stable-check` endpoint, can I:
- retrieve detailed information for that specific version through `version-check`/some other endpoint with similar information?
- specify which locale to check for?

**Comment by schlessera** [comment]
The call to the `stable-check` is really slow and often times out. Is that endpoint not being cached?

**Comment by schlessera** [comment]
@ocean90 Do you know whether there might be a caching issue with `stable-check` ^^^?

**Comment by ocean90** [comment]
There's no caching for that endpoint.

**Comment by schlessera** [comment]
> There's no caching for that endpoint.

Shouldn't that be changed, then? It seems slow and unreliable.

**Comment by schlessera** [comment]
@ocean90 Is there any update on this regarding the missing caching on the `stable-check` endpoint?
